transportable-udfs-utils/src/main/java/com/linkedin/transport/utils/FileSystemUtils.java

```diff
@@ -107,19 +113,23 @@ public static String resolveLatest(String path, FileSystem fs) throws IOExceptio
     }
   }
 
-  private static String resolveLatestHelper(String path, FileSystem fs) throws IOException {
+  private static String resolveLatestHelper(String path, FileSystem fs, boolean allowFile) throws IOException {
     if (!StringUtils.isBlank(path)) {
       path = path.trim();
       if (path.endsWith("/")) {
         path = path.substring(0, path.length() - 1);
       }
 
       FileStatus[] filesAndDirectories = fs.listStatus(new Path(path));
-      List<FileStatus> directories =
-          Arrays.stream(filesAndDirectories).filter(s -> s.isDirectory()).collect(Collectors.toList());
-      Collections.sort(directories);
-      if (directories != null && directories.size() != 0) {
-        String retval = path + "/" + directories.get(directories.size() - 1).getPath().getName();
+      List<FileStatus> candidates;
```

Comment:
Since I changed allowFile -> excludeFiles, here I changed to
`      List<FileStatus> candidates = Arrays.stream(filesAndDirectories).filter(s -> !excludeFiles || s.isDirectory())
          .sorted().collect(Collectors.toList());`